pkg/sourcepods/repository/service.go

```diff
@@ -103,10 +104,18 @@ func (s *service) Branches(ctx context.Context, owner, name string) ([]*Branch,
 
 	var branches []*Branch
 	for _, b := range bs {
+		var count int64
+		if b.Name == "master" {
+			count, err = s.storage.CommitCount(ctx, r.ID, b.Sha1)
```

Comment:
This is an N+1 :smile:

Reply:
Not sure if we want to return the commit-count in the branch-list... we could just leave it empty...